### ISSUE TITLE
Discussion regarding projects and classes

### ISSUE THREAD

**Issue by AmadeusW** [AmadeusW/Poker#5]
Let's talk about the classes and project that we will need for the project. Once that's figured out, we can create projects, unit tests, and start coding.

Here is an initial proposal:

Table
===
Table is the overarching entity that orchestrates the entire game.
There's definitely stuff that I've missed, for example, related to blinds
- List<IPlayer> Players
- IPlayer BigBlind
- IPlayer SmallBlind
- IDealer Dealer
- int CurrentBet
- int MinimumBet
- int MaximumBet
- int MaximumPlayers
- int MinimumPlayers
- int BuyIn
- List<Card> VisibleCards

Move
===
Move is what each IPlayer can do
- Fold
- Raise
- Check
- AllIn

IPlayer
===
Both human player and the AI implement the IPlayer interface. I believe that both GUI and AI code can be driven entirely by these methods:
- ObserveMove() - take note of other player's move, or take note of the entire table, which includes the cards that are visible.
- ObserveHand() - done at the end of the game
- MakeMove()
- List<Card> Hand
- RevealHand() - at the end of the game

IDealer
===
- Deal(IPlayer player) - deals a card to the player
- what else does a dealer do?

**Comment by Morkaleb** [comment]
Yeah, this is final product goals.  We can add an overarching video presentation to the overall game and really have something.  Some of this is a bit over my head currently I think.  My thinking is something like leading up to our next meeting, I would like to build a starting method, using what I know how to do, and see each other person build what they how to built within the project framework and improve/learn/put it all together from there.  What are the groups thoughts on this?

**Comment by AmadeusW** [comment]
> Yeah, this is final product goals.

Yup, that's the break down of what classes we need to have and what would be the relationship between them. This doesn't have any implementation, especially any of the AI implementation which will be the hardest part of the project. It's just the first step at the drafting board.

>  build what they how to built within the project framework

That would make the framework of the project. With this, everyone will know what to build, and it will be as easy as going down a to-do list.

I can start building this framework and guide us all through the process of building it.

**Comment by Morkaleb** [comment]
Alright, sounds good.  Going forward than, I'll continue writing code for the dealing.  We'll put what we can of it into use, and rewrite what needs to be rewritten.

**Comment by Morkaleb** [comment]
Actually, let me see what I can get setup within Unity.  I can at least work on setting up the necessary art.

**Comment by AmadeusW** [comment]
Okay, keep us posted!
Btw did I miss any of game elements in the writeup above?

**Comment by Morkaleb** [comment]
You haven't missed any game elements that I can think of that you haven't mentioned.  I agree that the final AI will be the most challenging  (and the most fun to build)  a simpler challenge to work towards that needs to be done first is the hand comparative method, since we have to build it to decide on the best 5 cards out of seven.  There are so many possibilities,  and the computer has to decide which 5 cards out of 7 to build each players hand with.  I have a couple of ideas, I am really excited about ttying them.  Ultimately this is harder than a simple but functional ai.  At least once this part is done.
